### PR TITLE
Introduce Measurement constructor from covariance and vector

### PR DESCRIPTION
**Pull request by asalzburger**
As discussed this morning, I encapsulated the problem from my work on the digitisation:
 * this PR will hopefully first fix the ambiguity in constructing parameter sets
 * then it should introduce a new `Measurement( ..., covariance, vector)` constructor

### CODE REVIEW

**Comment by asalzburger** [comment]
FYI @paulgessinger @FabianKlimpel @msmk0  

I did not change anything on the acts code, just added a test that should work:

The offening lines are those:
```c++
  // One-dimensional constructions
  ActsSymMatrixD<1> cov1D;
  cov1D << 0.5;

  ActsVectorD<1> vec1D;
  vec1D << 0.1;

  ParameterSet<BoundParametersIndices, eBoundLoc0> parSet_with_cov1D_real(cov1D,
                                                                          0.1);

  // This line does not compile
  ParameterSet<BoundParametersIndices, eBoundLoc0> parSet_with_cov1D_vec1D(
      cov1D, vec1D);
```

`vec1D` and `0.1` lead to an ambiguity in translating that into a parameter set.

**Comment by asalzburger** [comment]
I think one way out would not to template the `initializer` with the `value_type` :

```c++
typename first_value_type,
            typename... other_value_type
```

and deduce the type from the vector to be initialised.

**Comment by paulgessinger** [comment]
Right, I see. So, could you just remove the new overload in case the dimension is 1?

File: Core/include/Acts/EventData/Measurement.hpp [hunk 1]
```diff
@@ -122,7 +122,34 @@ class Measurement {
...
+  /// @brief standard constructor for surface/volume measurements
+  ///
+  /// Concrete class for all possible measurements, built from properly
+  /// formatted covariance matrix and parameter vector
+  ///
+  /// @note Only a reference to the given surface/volume is stored. The user
+  /// must ensure that the lifetime of the @c Surface / @c Volume object
+  /// surpasses the lifetime of this Measurement object. The given parameter
+  /// values are interpreted as values to the parameters as defined in the class
+  /// template argument @c params.
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/EventData/Measurement.hpp]: I removed those remarks

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**